### PR TITLE
Colinmacdonald/tests build readme

### PR DESCRIPTION
**Pull request by colinmacdonald**
- Adds project files
  - .gitignore
  - Makefile
  - LICENSE
  - component.json
  - package.json
- Adds README.md
- Adds tests and travis config
  - test/index.html
  - test/index.js
  - .travis.yml
- Removes bower from grunt build task

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/goinstant/usercache.git"
+  },
+  "author": "",
+  "readmeFilename": "README.md",
+  "bugs": {
+    "url": "https://github.com/goinstant/usercache/issues"
+  },
+  "devDependencies": {
```
Comment (JeremyWright) [inline package.json:18]: Needs a property for namespacing the repo.

ie for widgets:

``` JSON
"goinstant": {
  "namespace": "widgets"
}
```

This is how the deploy-service knows where to put it

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+language: node_js
+node_js:
+- '0.10'
+before_script:
+- npm install -g grunt-cli
+env:
+  global:
+  - secure: VORTuShe7FuzZLKJDLeRlHAKmxQtQd2k15jTFnebCrm/fpEwcwQddeb+T9rWMjirtRAxQnmhj+S6DqTJWIW4TOGI3SaDSvIpv2bTi9zwTyjUWsRHJXmTydJ0zaM6rhzvenZKpgOOGFB+hdbZvrtQS5gUS3z9dEyn30VpO5PgXbY=
```
Comment (JeremyWright) [inline .travis.yml:8]: What are these keys?
Are these safe to open-source @jbowes

File: package.json (hunk 1 above)
Comment (colinmacdonald) [inline package.json:18]: we aren't deploying this one as its not an official widget, its just a common module being used by other widgets. Just has to be visible to component.io when installing the other widgets.

File: .travis.yml (hunk 2 above)
Comment (colinmacdonald) [inline .travis.yml:8]: They are encrypted using the travis gem as per the SL tutorial: https://saucelabs.com/docs/javascript-unit-testing-tutorial
it encrypts our SL username and api key

File: package.json (hunk 1 above)
Comment (JeremyWright) [inline package.json:18]: Okie

File: .travis.yml (hunk 2 above)
Comment (JeremyWright) [inline .travis.yml:8]: K works for me

**Comment by JeremyWright** [comment]
Provided everything works/builds ok this looks good to me. @nealstewart anything else you'd like to add?

File: .travis.yml (hunk 2 above)
Comment (jbowes) [inline .travis.yml:8]: yeah this should be fine. especially for api keys like for saucelabs where if it gets compromised we can just regen a new one.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+[![Build Status](https://magnum.travis-ci.com/goinstant/usercache.png?token=yxZ3kgnG75WoPosksjKb&branch=master)](https://magnum.travis-ci.com/goinstant/usercache)
+
+## Usercache
+
+The Usercache component implements a synchronous interface for working with
+GoInstant users and is used in various [GoInstant widgets](https://developers.goinstant.com/v1/widgets/index.html).
+
+## Packaging
```
Comment (nealstewart) [inline README.md]: You can delete the packaging section. That really only makes sense for a public facing module.

Comment (nealstewart) [inline README.md]: We really only need Contributing

**Comment by nealstewart** [comment]
Is this repo public? It needs to be.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+[![Build Status](https://travis-ci.org/goinstant/goangular.png?branch=master)](https://travis-ci.org/goinstant/goangular)
```
Comment (nealstewart) [inline README.md]: I assume you didn't mean to make this goangular's build status?

Comment (colinmacdonald) [inline README.md]: oops

**Comment by colinmacdonald** [comment]
@nealstewart LGTM?

**Comment by nealstewart** [comment]
LGTM

**Pull request merged**